internal/helper/helper.go

```diff
@@ -201,64 +232,94 @@ func CheckEmptyFields(Interface interface{}, result *gojsonschema.Result, schema
-			tags := elem.Type().Field(i).Tag
-			jsonTag, hasJSON := tags.Lookup("json")
-			if hasJSON {
-				if !strings.Contains(jsonTag, "omitempty") {
-					required = true
-				}
+			tags, requiredFromTags := getTags(elem, i)
+			if !requiredFromTags && !isRequiredFromSchema(elem, i, schema) {
+				continue
 			}
-			// also check for required values in the jsonschema
-			for _, requiredField := range schema.Required {
-				if elem.Type().Field(i).Name == requiredField {
-					required = true
-				}
+			// this is a required field, check for zero values
+			if !reflect.Indirect(field).IsZero() {
+				continue
 			}
-			if required {
-				// this is a required field, check for zero values
-				if reflect.Indirect(field).IsZero() {
-					jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
-					errDetail := gojsonschema.ErrorDetails{
-						"property": tags.Get("yaml"),
-						"parent":   elem.Type().Name(),
-					}
-					result.AddError(
-						newMissingFieldError(
-							gojsonschema.NewJsonContext("missing_field", jsonContext),
-							52,
-							errDetail,
-						),
-						errDetail,
-					)
-				}
+			jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
+			errDetail := gojsonschema.ErrorDetails{
+				"property": tags.Get("yaml"),
+				"parent":   elem.Type().Name(),
 			}
+			result.AddError(
+				newMissingFieldError(
+					gojsonschema.NewJsonContext("missing_field", jsonContext),
+					52,
+					errDetail,
+				),
+				errDetail,
+			)
 		}
 	}
 	return nil
 }
 
+func checkEmptyFieldsInSlice(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	for i := 0; i < field.Cap(); i++ {
+		sliceElem := field.Index(i)
+		if sliceElem.Kind() == reflect.Ptr && sliceElem.Elem().Kind() == reflect.Struct {
+			err := CheckEmptyFields(sliceElem.Interface(), result, schema)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func checkEmptyFieldsInPtr(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	if field.Elem().Kind() == reflect.Struct {
+		err := CheckEmptyFields(field.Interface(), result, schema)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// getTags returns the json tags associated to the given element and checks if
+// these tags specify that the field is required.
+func getTags(elem reflect.Value, i int) (reflect.StructTag, bool) {
+	required := false
+	tags := elem.Type().Field(i).Tag
+	jsonTag, hasJSON := tags.Lookup("json")
+	if hasJSON {
+		if !strings.Contains(jsonTag, "omitempty") {
+			required = true
+		}
+	}
+
+	return tags, required
+}
+
+// isRequiredFromSchema checks if the field is required from the schema
+func isRequiredFromSchema(elem reflect.Value, i int, schema *jsonschema.Schema) bool {
+	required := false
+	// also check for required values in the jsonschema
+	for _, requiredField := range schema.Required {
+		if elem.Type().Field(i).Name == requiredField {
+			required = true
```

Comment:
You could simply return true here. Once true is assigned you don't change it anymore so there is not reason to continue looping.

Reply:
Indeed. I was a little to quick in fixing this. I have now 2 functions to if the field is required. I could gather everything under a single one but I like them independents because these are 2 different checks.